File: presentations/14-repository-and-builder-pattern-demos/csharp-smart-home-scenes/Domain/SceneTarget.cs
namespace SmartHomeScenesDemo.Domain;

public enum SceneTargetKind
{
    SpecificDevice,
    DeviceGroup
}

public abstract record SceneTarget(SceneTargetKind Kind);

public sealed record SpecificDeviceTarget(Guid DeviceId)
    : SceneTarget(SceneTargetKind.SpecificDevice);

public sealed record DeviceGroupTarget(string DeviceType, string? Location)
    : SceneTarget(SceneTargetKind.DeviceGroup);

File: presentations/14-repository-and-builder-pattern-demos/java-smart-home-scenes/src/main/java/edu/ksu/swe4743/smarthomescenes/domain/SceneTarget.java
package edu.ksu.swe4743.smarthomescenes.domain;

public abstract class SceneTarget {
    private final SceneTargetKind kind;

    protected SceneTarget(SceneTargetKind kind) {
        this.kind = kind;
    }

    public SceneTargetKind getKind() {
        return kind;
    }
}
